pkg/utils/utils.go

```diff
@@ -180,3 +202,75 @@ func GetVFLinkNames(pfName string, vfID int) ([]string, error) {
 
 	return names, nil
 }
+
+// HasDpdkDriver checks if a device is attached to dpdk supported driver
+func HasDpdkDriver(pciAddr string) (bool, error) {
+	driverLink := filepath.Join(SysBusPci, pciAddr, "driver")
+	driverPath, err := filepath.EvalSymlinks(driverLink)
+	if err != nil {
+		return false, err
+	}
+	driverStat, err := os.Stat(driverPath)
+	if err != nil {
+		return false, err
+	}
+	driverName := driverStat.Name()
+	for _, drv := range UserspaceDrivers {
+		if driverName == drv {
+			return true, nil
+		}
+	}
+	return false, nil
+}
+
+// SaveNetConf takes in container ID, data dir and Pod interface name as string and a json encoded struct Conf
+// and save this Conf in data dir
+func SaveNetConf(cid, dataDir, podIfName string, conf interface{}) error {
+	netConfBytes, err := json.Marshal(conf)
+	if err != nil {
+		return fmt.Errorf("error serializing delegate netconf: %v", err)
+	}
+
+	s := []string{cid, podIfName}
+	cRef := strings.Join(s, "-")
+
+	// save the rendered netconf for cmdDel
+	if err = saveScratchNetConf(cRef, dataDir, netConfBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func saveScratchNetConf(containerID, dataDir string, netconf []byte) error {
+	if err := os.MkdirAll(dataDir, 0700); err != nil {
+		return fmt.Errorf("failed to create the sriov data directory(%q): %v", dataDir, err)
```

Comment:
This might be a problem on system which has immutable filesystem (e.g openshift), perhaps make the dataDir (DefaultCNIDir) configurable?

Reply:
You mean we should retain the json 'cniDir' so user can choose the directory in config file?

Reply:
I think using the default dir shall be fine for now. will file an issue if it doesn't work later; mark it as resolved.